src/tools/spin/spinquic.cpp

```diff
@@ -537,6 +545,20 @@ void Spin(LockableVector<HQUIC>& Connections, std::vector<HQUIC>* Listeners = nu
             }
             break;
         }
+        case SpinQuicAPICallStreamReceiveComplete: {
+            auto Connection = Connections.TryGetRandom();
+            BAIL_ON_NULL_CONNECTION(Connection);
+            auto ctx = SpinQuicConnection::Get(Connection);
+            {
+                std::lock_guard<std::mutex> Lock(ctx->Lock);
+                auto Stream = ctx->TryGetStream();
+                if (Stream == nullptr) continue;
+                auto BytesRemaining = MsQuic->GetContext(Stream);
+                MsQuic->StreamReceiveComplete(Stream, (uint64_t)BytesRemaining);
```

Comment:
You should (only sometimes) complete less than the bytes remaining, which will also pause.